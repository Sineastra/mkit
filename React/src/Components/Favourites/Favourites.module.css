.wrapper {
  background: var(--theme-main-color_2);
  color: var(--theme-text-color);
  padding: 30px 50px;
}
.wrapper > h1 {
  text-align: center;
  margin: 0 0 50px;
}
.wrapper .empty {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
}
.wrapper .nonEmpty {
  display: grid;
  grid-template-columns: repeat(auto-fit, 200px);
  column-gap: 30px;
}

/*# sourceMappingURL=Favourites.module.css.map */
