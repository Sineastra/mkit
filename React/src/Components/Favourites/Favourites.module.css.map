{"version":3,"sourceRoot":"","sources":["Favourites.module.scss","../../_scssPartials/_mixins.scss"],"names":[],"mappings":"AAEA;EACI;EACA;EACA;;AAEA;EACI;EACA;;AAIJ;ECZA;EACA,gBAM4B;EAL5B,iBAMiC;EALjC,aAKyC;EAJzC,eAIiD;;ADQjD;EACI;EACA;EACA","file":"Favourites.module.css"}