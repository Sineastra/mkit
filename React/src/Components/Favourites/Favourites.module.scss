@import "../../_scssPartials/mixins";

.wrapper {
    background: var(--theme-main-color_2);
    color: var(--theme-text-color);
    padding: 30px 50px;

    & > h1 {
        text-align: center;
        margin: 0 0 50px;
    }


    & .empty {
        @include centeredFlex()
    }

    & .nonEmpty {
        display: grid;
        grid-template-columns: repeat(auto-fit, 200px);
        column-gap: 30px;
    }
}