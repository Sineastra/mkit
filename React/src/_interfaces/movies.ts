export interface movieIntF {
	_id?: string,
	externalId: string,
	poster: string,
	title: string,
	genres: string[],
	runtime: number,
	officialSite: string,
	description: string,
}

export interface movieDetailsForUserIntF {
	rating: number
	notes: string
}