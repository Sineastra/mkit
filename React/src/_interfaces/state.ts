export interface movieInteractionIntF {
	_id: string,
	rating: number,
	privateComment: string,
}
export interface userData {
	_id: string,
	name: string,
	iat?: number,
}
export interface userSliceIntF {
	loggedUser: userData | null
}