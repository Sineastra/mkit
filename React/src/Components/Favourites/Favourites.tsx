import { FavCardMovieIntF } from "../../_interfaces/movies"
import styles from "./Favourites.module.scss"
import SingleGridMovie from "../FavCard/FavCard"


interface propsIntF {
	movies: FavCardMovieIntF[]
}
const Favourites = ({ movies }: propsIntF) => {

	return (
		<section className={ styles.wrapper }>
			<h1>My Favourites</h1>
			{ movies.length === 0
				? <div className={ styles.empty } role="no-fav-msg">
					You don't have any favourites yet. Go and browse like a browser.
				</div>
				: <div className={ styles.nonEmpty } role="fav-grid">
					{ movies.map(x => <SingleGridMovie movie={ x } key={ x.imdbId }/>) }
				</div>
			}
		</section>
	)
}

export default Favourites