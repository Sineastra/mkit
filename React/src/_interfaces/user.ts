import { movieIntF } from "./movies"


export interface user {
	_id: string,
	name: String,
	hashedPassword: String,
	favourites: movieIntF[],
	notes: { movieId: string, note: string }[],
	ratings: { movieId: string, rating: string }[],
}
