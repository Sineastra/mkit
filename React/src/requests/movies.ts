import { FavCardMovieIntF, SearchCardMovieIntF } from "../_interfaces/movies"
import { baseAPIRequest } from "./baseRequest"


const getDetailedInfo = (movie: any): SearchCardMovieIntF => {
	return {
		imdbId: movie.externals.imdb,
		poster: movie.image.medium,
		title: movie.name,
		genres: movie.genres,
		runtime: movie.runtime,
		officialSite: movie.officialSite,
		description: movie.summary,
	}
}

interface moviesRequestsIntF {
	getFavs: (ids: string[]) => Promise<FavCardMovieIntF[]>,
	getSearchs: (query: string) => Promise<SearchCardMovieIntF[]>,
	getDetails: (id: string) => Promise<SearchCardMovieIntF>
}
const movieRequests: moviesRequestsIntF = {
	getFavs: async (ids) => {
		const fetchReq = async (id: string): Promise<FavCardMovieIntF> => {
			const movieData = await baseAPIRequest(`/lookup/shows?imdb=${ id }`)

			return {
				imdbId: movieData.externals.imdb,
				poster: movieData.image.medium,
				title: movieData.name,
				genres: movieData.genres,
			}
		}
		const promises = ids.map(fetchReq)
		const movies: FavCardMovieIntF[] = await Promise.all(promises)

		return movies
	},
	getSearchs: async (query) => {
		const movieData = await baseAPIRequest(`/search/shows?q=${ query }`)
		const movies: SearchCardMovieIntF[] = movieData.map((x: any) => {
			return getDetailedInfo(x)
		})

		return movies
	},
	getDetails: async (id) => {
		const movieData = await baseAPIRequest(`/lookup/shows?imdb=${ id }`)

		return getDetailedInfo(movieData)
	},
}

export { movieRequests }