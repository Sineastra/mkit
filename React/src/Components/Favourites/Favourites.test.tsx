import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom"
import Favourites from "./Favourites"
import { movieIntF } from "../../_interfaces/movies"


const renderTree = (movies: movieIntF[]) =>
	renderer.create(
		<BrowserRouter>
			<Favourites favs={ movies }/>
		</BrowserRouter>,
	)
const mockedMovies: movieIntF[] = [{
	externalId: 'a',
	poster: 'b',
	title: 'c',
	genres: ['d', 'e'],
	runtime: 1,
	officialSite: 'b',
	description: 'b',
}, {
	externalId: 'd',
	poster: 'c',
	title: 'b',
	genres: ['a', 'b'],
	runtime: 1,
	officialSite: 'b',
	description: 'b',
}]

describe("---> Testing /Components/Favourites", () => {
	it("should render correctly when there are no favourite movies", () => {
		const tree = renderTree(mockedMovies).toJSON()

		expect(tree).toMatchSnapshot()
	})
	it("should render correctly when there are NO favourite movies", () => {
		const tree = renderTree([]).toJSON()

		expect(tree).toMatchSnapshot()
	})
})