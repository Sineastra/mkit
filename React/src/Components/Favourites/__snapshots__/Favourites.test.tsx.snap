// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---> Testing /Components/Favourites should render correctly when there are NO favourite movies 1`] = `
<section
  className="wrapper"
>
  <h1>
    My Favourites
  </h1>
  <div
    className="empty"
    role="no-fav-msg"
  >
    You don't have any favourites yet. Go and browse like a browser.
  </div>
</section>
`;

exports[`---> Testing /Components/Favourites should render correctly when there are no favourite movies 1`] = `
<section
  className="wrapper"
>
  <h1>
    My Favourites
  </h1>
  <div
    className="nonEmpty"
    role="fav-grid"
  >
    <div
      role="slide"
    >
      <a
        className="link"
        href="/details/a"
        onClick={[Function]}
      >
        <img
          alt="poster"
          className="slide"
          src="b"
        />
        <h4
          role="title"
        >
          c
        </h4>
        <h5
          role="genre"
        >
          d, e
        </h5>
      </a>
    </div>
    <div
      role="slide"
    >
      <a
        className="link"
        href="/details/d"
        onClick={[Function]}
      >
        <img
          alt="poster"
          className="slide"
          src="c"
        />
        <h4
          role="title"
        >
          b
        </h4>
        <h5
          role="genre"
        >
          a, b
        </h5>
      </a>
    </div>
  </div>
</section>
`;
